src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/service/PrisonerContactRegistryService.kt

```diff
@@ -111,6 +125,33 @@ class PrisonerContactRegistryService(private val prisonApiClient: PrisonApiClien
     return dateRange
   }
 
+  @Throws(VisitorNotFoundException::class)
+  fun getClosedRestrictionStatusForPrisonerContacts(prisonerId: String, visitorIds: List<Long>): HasClosedRestrictionDto {
+    log.debug(
+      "getHasClosedRestrictionForPrisonerContacts called with parameters : prisonerId - {}, visitorIds - {}",
+      prisonerId,
+      visitorIds,
+    )
+
+    val hasClosedRestrictionDto = HasClosedRestrictionDto(false)
+
+    val visitors = getVisitors(prisonerId, visitorIds)
+    val visitorClosedRestrictions =
+      visitors
+        .flatMap { it.restrictions }
+        .filter { it.restrictionType == CLOSED_RESTRICTION_TYPE }
+
+    visitorClosedRestrictions.forEach { restriction ->
+      restriction.expiryDate?.let { expiryDate ->
+        if (LocalDate.now() <= expiryDate) {
+          hasClosedRestrictionDto.value = true
+        }
+      } ?: run { hasClosedRestrictionDto.value = true }
+    }
```

Comment:
suggestion only - can be replaced with restrictions.any{check} to avoid looping through all